File: programmers/연습문제/콜라츠추측.py
def solution(num):
    answer = 0
    while num != 1:
        if answer > 500:
            answer = -1
            break
        if num % 2: # 홀수인 경우
            num = num * 3 + 1
        else: # 짝수인 경우
            num = int(num // 2)
        answer += 1
    return answer

File: programmers/연습문제/콜라츠 추측.kt
class Solution {
    fun solution(n: Int): Int {
        var answer = 0
        var n = n.toLong()
        while (n != 1.toLong()) {
            answer += 1
            if (answer > 500) {
                answer = -1
                break
            }
            n = if (n % 2 == 0.toLong()) n / 2 else n * 3 + 1
        }
        return answer
    }
}